### ISSUE TITLE
Add option for text top-edge and bottom-edge to "shrinkwrap" text

### ISSUE THREAD

**Issue by T0mstone** [typst/typst#469]
Currently it is only possible to set `text(top-edge: "ascender" / "cap-height" / ...)` in a fixed way,
but I would like an option that always makes the text's layout box as large as the text actually is.

If I write
```tex
\fbox{aa} \fbox{aA}
```
in LaTeX, I get
[texbox.pdf](https://github.com/typst/typst/files/11115594/texbox.pdf)

On the other hand, something like
```typst
#set page(width: 6em, height: 2em)

#stack(
	dir: ltr,
	rect[aa],
	h(1em),
	rect[aA]
)
```
gives me
[box.pdf](https://github.com/typst/typst/files/11115602/box.pdf)

I don't think there is currently any workaround for this, so I'd propose adding `"shrinkwrap"` (or whatever it would be called) as a possible value for `top-edge`/`bottom-edge` in `text`.

**Comment by Leedehai** [comment]
It was mentioned on the Discord question https://discord.com/channels/1054443721975922748/1108195933784637441 which looks like a pretty significant layout issue.

Also, for readers' convenience I took screenshots of the PDF files above (clicking those links would trigger file downloading):
textbox.pdf: 
<img width="56" alt="Screenshot 2023-05-17 at 23 09 36" src="https://github.com/typst/typst/assets/18319900/a0435e84-469b-4784-899c-2237f8d8186b">
box.pdf:
<img width="98" alt="Screenshot 2023-05-17 at 23 09 54" src="https://github.com/typst/typst/assets/18319900/48cac186-6725-42e6-8197-5e49247bcfaf">

**Comment by laurmaedje** [comment]
The math issue is related but at the same time somewhat separate, there is also an issue for it here: #1028

**Comment by LU15W1R7H** [comment]
I would also argue that this new option (if needed) should become the new default. You never want overlapping text, do you?

**Comment by Enivex** [comment]
> I would also argue that this new option (if needed*) should become the new default. You never want overlapping text, do you?

It should absolutely not be the default, because it would lead to uneven line spacing. There's a good reason for why the bottom and top edge aren't overly sensitive to the actual text.

**Comment by sky126** [comment]
When will this problem be fixed?

**Comment by T0mstone** [comment]
Closed by #1626
